File: src/ui.cpp
#include "ui.hpp"

#include <iostream>
#include <cstdlib>

void UI::start(Window window) {
    set_window(window);
    get_instance().run();
}

void UI::set_window(Window window) {
    get_instance().current_window = window;
}

void UI::stop() {
    get_instance().running = false;
}

UI::UI() : running(true) {}
UI::~UI() {}

UI &UI::get_instance() {
    static UI instance;
    return instance;
}

void UI::run() {
    while (running) {
        current_window();

        std::cout << std::endl << "############################" << std::endl << std::endl;
    }
}

File: src/ui/UI.java
package ui;

import util.ANSI;

public class UI {
    private static UI INSTANCE = new UI();
    private boolean running;
    private Runnable currentWindow;

    private UI() {
        this.running = true;
    }

    public static void start(Runnable window) {
        setWindow(window);
        INSTANCE.run();
    }

    public static void setWindow(Runnable window) {
        INSTANCE.currentWindow = window;
    }

    public static void stop() {
        INSTANCE.running = false;
    }

    private void run() {
        while (running) {
            if (currentWindow != null) {
                currentWindow.run();
            }

            System.out.println(ANSI.BLUE + "\n############################\n" + ANSI.RESET);
        }
    }
}